homeassistant/components/modbus/modbus.py

```diff
@@ -293,7 +293,7 @@ def write_coil(self, unit, address, value) -> bool:
                 result = self._client.write_coil(address, value, **kwargs)
             except ModbusException as exception_error:
                 result = exception_error
-            if not hasattr(result, "registers"):
+            if not hasattr(result, "value"):
```

Comment:
Very good catch, I just checked with the modbus standard and even though I do not like to admit it 😁 you are correct.

Reply:
I didn't care about the standard, but I looked at the structures returned by PyModbus. Whatever works for us :)